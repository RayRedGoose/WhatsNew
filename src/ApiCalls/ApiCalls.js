export const getArticles = () => {
  return fetch('https://whats-new-api.herokuapp.com/api/v1/news')
    .then(response => {
      if (!response.ok) {
        throw Error('Error fetching articles')
      }
      return response.json()
    })
}
