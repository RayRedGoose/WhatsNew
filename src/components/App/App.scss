* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
  display: grid;
  grid-template-areas: "menu search" "menu main";
  grid-template-columns: 1fr 4fr;
  grid-template-rows: 1fr 9fr;
  background: #141527;
  height: 100vh;
  text-align: center;
  overflow: hidden;
  width: 100%;
}

@media only screen and (max-width: 600px) {
  .app {
    display: flex;
    flex-direction: column;
    height: 100%;
  }
}
