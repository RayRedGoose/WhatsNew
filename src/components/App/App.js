import React, { Component } from 'react';
import './App.scss';
import Menu from '../Menu/Menu.js'
import SearchForm from '../SearchForm/SearchForm'
import LoadingImage from '../LoadingImage/LoadingImage'
import NewsContainer from '../NewsContainer/NewsContainer'
import { getArticles } from '../../ApiCalls/ApiCalls'

class App extends Component {
  constructor() {
    super();
    this.state = {
      activeTab: null
    }
  }

  componentDidMount() {
    getArticles()
      .then(
        (result) => {
          this.setState({
            ...result,
            isLoaded: true,
            activeTab: "local"
          });
        }
      )
      .catch(err => console.error(err))
  }

  changeArticles = (info) => {
    this.setState({activeTab: info})
  }

  render () {
    const articlesOnBoard = (typeof this.state.activeTab === 'string')
      ? this.state[this.state.activeTab]
      : this.state.activeTab

    if (this.state.isLoaded) {
      return (
        <div className="app">
          <Menu changeArticleTheme={this.changeArticles} />
          <SearchForm
            changeArticleSwown={this.changeArticles}
            articleShown={articlesOnBoard}
          />
          <NewsContainer newsArticles={articlesOnBoard} />
        </div>
      )
    }
    if (!this.state.isLoaded) {
      return (
        <div className="app loading">
          <LoadingImage />
        </div>
      )
    }
  }
}

export default App;
