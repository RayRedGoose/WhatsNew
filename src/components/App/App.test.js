import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { getArticles } from '../../ApiCalls/ApiCalls';

describe('App', () => {
  let container;

  beforeEach(() => {
    container = shallow(<App />);
  })

  it('should match the snapshot with no data fetched', () => {
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot with all data fetched', () => {
    container.setState({ isLoaded: true });
    expect(container).toMatchSnapshot();
  });

  it('should update state with an active tab when changeArticles is called', () => {
    // Setup
    const activeTab = 'local';
    const expected = activeTab;

    // Expectation
    expect(container.state('activeTab')).toEqual(null);

    // Execution
    container.instance().changeArticles(activeTab);

    // Expectation
    expect(container.state('activeTab')).toEqual(expected);
  });
});
