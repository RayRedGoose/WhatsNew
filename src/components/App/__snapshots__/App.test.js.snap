// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data fetched 1`] = `
<div
  className="app"
>
  <Menu
    changeArticleTheme={[Function]}
  />
  <SearchForm
    articleShown={null}
    changeArticleSwown={[Function]}
  />
  <NewsContainer
    newsArticles={null}
  />
</div>
`;

exports[`App should match the snapshot with no data fetched 1`] = `
<div
  className="app loading"
>
  <LoadingImage />
</div>
`;
